pmlc/dialect/tile/contraction.cc

```diff
@@ -263,6 +263,50 @@ static IndexPoly MakePoly(mlir::Value* value) {
   throw std::runtime_error("Invalid affine op");
 }
 
+static IndexPoly MakePoly(mlir::AffineExpr expr) {
+  IVLOG(3, "MakePoly: " << mlir::debugString(expr));
+  if (auto dimExpr = expr.dyn_cast<mlir::AffineDimExpr>()) {
+    auto name = llvm::formatv("x{0}", dimExpr.getPosition());
+    return IndexPoly{name.str()};
+  }
+  if (auto constExpr = expr.dyn_cast<mlir::AffineConstantExpr>()) {
+    return IndexPoly{constExpr.getValue()};
+  }
+  if (auto binaryExpr = expr.dyn_cast<mlir::AffineBinaryOpExpr>()) {
+    switch (binaryExpr.getKind()) {
+      case mlir::AffineExprKind::Add:
+        return MakePoly(binaryExpr.getLHS()) + MakePoly(binaryExpr.getRHS());
+      case mlir::AffineExprKind::Mul: {
+        auto rhs = MakePoly(binaryExpr.getRHS());
+        return MakePoly(binaryExpr.getLHS()) * rhs.constant();
+      }
+      case mlir::AffineExprKind::FloorDiv: {
+        auto rhs = MakePoly(binaryExpr.getRHS());
+        return MakePoly(binaryExpr.getLHS()) / rhs.constant();
```

Comment:
Should we be checking that the RHS poly is constant? Or is it ok to be assuming that safe polynomials were passed in? (same question on * case)

Reply:
OK I added a check for this